### ISSUE TITLE
@typescript-eslint/no-unused-vars false positive

### ISSUE THREAD

**Issue by mmussmann** [typescript-eslint/typescript-eslint#2507]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
 "extends": [
    "eslint:recommended",
    "plugin:react/recommended",
    "plugin:react-hooks/recommended",
    "plugin:@typescript-eslint/recommended",
    "prettier"
  ],

  "plugins": ["react"],

  "parser": "@typescript-eslint/parser",

  "rules": {
    "@typescript-eslint/explicit-module-boundary-types": "off"
  }
```

```TS
type StyledTreeProps = {
  hide: boolean;
};
const StyledTree = styled.div<StyledTreeProps>`
  line-height: 25px;
  margin-bottom: 8px;
  display: ${(props) => (props.hide ? 'none' : 'block')};
`;
```

**Expected Result**
No lint error

**Actual Result**
  7:6  warning  'StyledTreeProps' is defined but never used  @typescript-eslint/no-unused-vars

**Additional Info**

**Versions**

| package                                   | version |
| ----------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin-tslint` | `4.0.1` |
| `@typescript-eslint/parser`               | `4.0.1` |
| `TypeScript`                              | `4.0.2` |
|`eslint`|`7.8.1`|
| `node`                                    | `12.18.3` |
| `styled-components`                                    | `5.2.0` |
| `@types/styled-components`                                    | `5.1.3` |

**Comment by bradzacher** [comment]
Duolicate of https://github.com/typescript-eslint/typescript-eslint/issues/2452

Please use the search.

In addition please read the template. You used the wrong template.
